Source/CombatExtended/CombatExtended/StatWorkers/StatWorker_Magazine.cs

```diff
@@ -26,7 +26,9 @@ public override bool ShouldShowFor(StatRequest req)
         }
 
         public override float GetValueUnfinalized(StatRequest req, bool applyPostProcess = true)
-        {
+        {
+            if ((req.Thing?.TryGetComp<CompAmmoUser>()?.Props?.ammoSet ?? null) != (((CompProperties_AmmoUser)req.Thing?.def?.comps?.Find(x => x is CompProperties_AmmoUser))?.ammoSet ?? null))

```

Comment:
Same as the other PR  TryGetComp is expensive, and should be cached whenever possible.